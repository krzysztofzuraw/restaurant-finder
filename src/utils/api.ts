import axios from 'axios';

import { configService } from '~src/services';

const config = configService.getConfig();

export const agentInstance = axios.create({ baseURL: config.GEOCODING_URL });
