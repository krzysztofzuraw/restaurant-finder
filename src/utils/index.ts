export * from './tests';
export * from './store';
export * from './api';
