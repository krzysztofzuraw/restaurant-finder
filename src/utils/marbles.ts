import { ActionsObservable, Epic, StateObservable } from 'redux-observable';
import { Subject } from 'rxjs';

import { TestScheduler } from 'rxjs/testing';
import Types from 'Types';
import { rootReducer } from '~src/store';

const testScheduler = new TestScheduler((actual, expected) => expect(actual).toEqual(expected));
const initialState = rootReducer(undefined, {} as Types.RootAction);

export const marbleTest = (
  actions: string,
  expected: string,
  values: { [marble: string]: Types.RootAction },
  services: {},
  epic: Epic
) =>
  testScheduler.run(({ cold, expectObservable }) => {
    const state$ = new StateObservable(new Subject<Types.RootState>(), initialState);
    const action$ = new ActionsObservable(cold<Types.RootAction>(actions, values));
    const dependencies = {
      geocodingService: {
        searchForPlaces: jest.fn(),
      },
      loggingService: {
        logError: jest.fn(),
      },
      configService: {
        getConfig: jest.fn(),
      },
      mapsService: {
        initMap: jest.fn(),
        placeCurrentLocationMarker: jest.fn(),
      },
      geolocationService: {
        getCurrentPosition: jest.fn(),
      },
      ...services,
    };
    const output$ = epic(action$, state$, dependencies);

    expectObservable(output$).toBe(expected, values);
  });
