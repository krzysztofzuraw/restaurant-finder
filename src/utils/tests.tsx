import * as React from 'react';

import { AppTheme } from '~src/theme';

export const setupComponent = (children: JSX.Element) => <AppTheme>{children}</AppTheme>;
