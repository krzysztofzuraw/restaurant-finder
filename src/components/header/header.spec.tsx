import * as React from 'react';
import { cleanup, render } from 'react-testing-library';

import { setupComponent } from '~src/utils';
import { Header } from '.';

afterEach(cleanup);

describe('Header Component', () => {
  it('should render correctly', () => {
    const component = render(setupComponent(<Header onSubmit={jest.fn()} onChange={jest.fn()} />));
    expect(component).toMatchSnapshot();
  });
});
