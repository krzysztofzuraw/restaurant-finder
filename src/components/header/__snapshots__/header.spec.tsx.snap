// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  background-color: #2470d8;
  grid-area: 'header';
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 h1 {
  font-size: 2.5em;
  color: #FFFFFF;
  font-weight: 400;
}

.emotion-0 {
  width: 351px;
  height: 44px;
  border-radius: 4px;
  border: none;
  font-size: 1em;
  line-height: 25px;
  background-image: url();
  background-repeat: no-repeat;
  background-position: 7px;
  padding-left: 30px;
  padding-right: 10px;
}

<body>
    <div>
      <div
        class="emotion-1"
      >
        <h1>
          Find a restaurant
        </h1>
        <form
          role="search"
        >
          <input
            class="emotion-0"
            placeholder="Type a name of restaurant..."
            type="search"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1jhrc66"
    >
      <h1>
        Find a restaurant
      </h1>
      <form
        role="search"
      >
        <input
          class="css-1ckarqe"
          placeholder="Type a name of restaurant..."
          type="search"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
